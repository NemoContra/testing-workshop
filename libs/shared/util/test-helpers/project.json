{
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util/test-helpers/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/shared/util/test-helpers/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/shared/util/test-helpers"],
      "options": {
        "jestConfig": "libs/shared/util/test-helpers/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["scope:shared", "type:util"]
}
