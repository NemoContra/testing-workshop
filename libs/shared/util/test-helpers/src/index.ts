export * from './lib/test-id-selector';
export * from './lib/create-spy-observer';
