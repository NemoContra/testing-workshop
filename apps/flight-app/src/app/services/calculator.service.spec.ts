import { AddService, CalculatorService } from './calculator.service';
import { of } from 'rxjs';
import { createSpyObserver } from '@testing-workshop/shared/util/test-helpers';

describe('CalculatorService', () => {
  let addService: AddService;
  let calculatorService: CalculatorService;

  beforeEach(() => {
    addService = new AddService();
    jest.spyOn(addService, 'calculate').mockReturnValue(of(1));
    calculatorService = new CalculatorService(addService);
  });

  it.each`
    nr1         | nr2
    ${1}        | ${1}
    ${-1}       | ${-1}
    ${1000000}  | ${2000000}
    ${-1000000} | ${-2000000}
  `('should add $nr1 and $nr2', ({ nr1, nr2 }) => {
    const spyOberserver = createSpyObserver();

    calculatorService.add(nr1, nr2).subscribe(spyOberserver);
    expect(spyOberserver).toMatchSnapshot();
  });

  it('should throw an error for negative values', () => {
    const spyOberserver = createSpyObserver();

    calculatorService.addUnsigned(-1, -2).subscribe(spyOberserver);
    expect(spyOberserver).toMatchSnapshot();
  });
});
