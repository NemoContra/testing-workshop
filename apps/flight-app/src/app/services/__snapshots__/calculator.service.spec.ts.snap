// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorService should add -1 and -1 1`] = `
Object {
  "complete": [MockFunction complete] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "error": [MockFunction error],
  "next": [MockFunction next] {
    "calls": Array [
      Array [
        1,
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;

exports[`CalculatorService should add -1000000 and -2000000 1`] = `
Object {
  "complete": [MockFunction complete] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "error": [MockFunction error],
  "next": [MockFunction next] {
    "calls": Array [
      Array [
        1,
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;

exports[`CalculatorService should add 1 and 1 1`] = `
Object {
  "complete": [MockFunction complete] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "error": [MockFunction error],
  "next": [MockFunction next] {
    "calls": Array [
      Array [
        1,
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;

exports[`CalculatorService should add 1000000 and 2000000 1`] = `
Object {
  "complete": [MockFunction complete] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "error": [MockFunction error],
  "next": [MockFunction next] {
    "calls": Array [
      Array [
        1,
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;

exports[`CalculatorService should throw an error for negative values 1`] = `
Object {
  "complete": [MockFunction complete],
  "error": [MockFunction error] {
    "calls": Array [
      Array [
        [Error: Both numbers must be greater than zero],
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "next": [MockFunction next],
}
`;
